jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/ButtonTests.java

```diff
@@ -27,138 +27,77 @@ public void before() {
-        iconButtonIcons.get(2).is().displayed();
-
-        iconButtons.get(3).click();
-        iconButtons.get(5).is().text("Last click: Alarm");
-        iconButtonIcons.get(3).is().displayed();
-
-        iconButtons.get(4).click();
-        iconButtons.get(5).is().text("Last click: Shopping cart");
-        iconButtonIcons.get(4).is().displayed();
+    @Test (dataProvider = "iconButtonTestData", dataProviderClass = ButtonDataProvider.class)
+    public void iconButtonTest(int buttonIndex, boolean isEnable) {
+        MUIButton iconButton = iconButtons.get(buttonIndex);
+        iconButton.icon().is().displayed();
+        if (isEnable) {
+            iconButton.is().enabled();
+            iconButton.click();
+        } else {
+            iconButton.is().disabled();
+        }
     }
 
-    @Test
-    public void customizedButtonTest() {
-
-        customizedButtons.get(1).click();
-        customizedButtons.get(1).is().text("CUSTOM CSS");
-        customizedButtons.get(4).is().text("Last click: Custom CSS");
-
-        customizedButtons.get(2).click();
-        customizedButtons.get(2).is().text("THEME PROVIDER");
-        customizedButtons.get(4).is().text("Last click: Theme Provider");
-
-        customizedButtons.get(3).click();
-        customizedButtons.get(3).is().text("Bootstrap");
-        customizedButtons.get(4).is().text("Last click: Bootstrap");
+    @Test (dataProvider = "customizeButtonTestData", dataProviderClass = ButtonDataProvider.class)
+    public void customizeButtonTest(int buttonIndex, String buttonTitle, boolean isEnable) {
+        MUIButton customizedButton = customizedButtons.get(buttonIndex);
+        customizedButton.label().is().text(buttonTitle);
+        if (isEnable) {
+            customizedButton.is().enabled();
+            customizedButton.click();
+        } else {
+            customizedButton.is().disabled();
+        }
     }
 
-    @Test
-    public void complexButtonTest() {
-
-        complexButtons.get(1).click();
-        complexButtons.get(1).is().text("Breakfast");
-        complexButtons.get(4).is().text("Last click:Breakfast");
-
-        complexButtons.get(2).click();
-        complexButtons.get(2).is().text("Burgers");
-        complexButtons.get(4).is().text("Last click:Burgers");
-
-        complexButtons.get(3).click();
-        complexButtons.get(3).is().text("Camera");
-        complexButtons.get(4).is().text("Last click:Camera");
+    @Test (dataProvider = "complexButtonTestData", dataProviderClass = ButtonDataProvider.class)
+    public void complexButtonTest(int buttonIndex, String buttonTitle, boolean isEnable) {
+        ComplexMUIButton complexButton = complexButtons.get(buttonIndex);
+        complexButton.typography().is().text(buttonTitle);
+        complexButton.image().is().displayed();
+        if (isEnable) {
+            complexButton.is().enabled();
+            complexButton.click();
+        } else {
+            complexButton.is().disabled();
+        }
```

Comment:
Is it really necessary to perform all the same type of checks? Maybe it's enough to check only one button?